### DISCUSSION TITLE
Outline what filters are active?

### DISCUSSION THREAD

**Discussion by Hipshot-zz** [Garux/netradiant-custom#185] [Ideas]
This menu here:
![bild 1](https://github.com/Garux/netradiant-custom/assets/178820/25b3d776-bb4f-42b1-a4ce-f0abae707ee6)

Right now, I don't know what item is selected here, expcept for that I could see it (see: not see it), in the editor viewports directly. Maybe a checkbox or something can be added to this menu?

Personally for me: Get rid of the icons and add checkboxes instead. But I can see that it looks nicer this way =)

**Comment by Garux** [comment]
Supposed to look this way with default theme. It inherits OS appearence.
![image](https://github.com/Garux/netradiant-custom/assets/1734117/07cfb657-e834-4c29-ac8c-35e0e64c3407)

**Comment by Hipshot-zz** [comment]
Actually, there is a checkbox thingie there. It's just that it's UNDER the icons... because if you filter some of these things with no icon, some of the things you would never care to filter, it marks it with a checkbox =) That checkbox could really be above the icons.

Not really sure why the blue outline isn't seen for me, not sure why that would disappear, can't figure out what it's with my OS that would make that different (w11).

**Comment by Garux** [comment]
No, it's concrete option on what to display <https://doc.qt.io/qt-5/qaction.html#iconVisibleInMenu-prop>.

Do toolbar toggle buttons have state indication in your OS? This may be bug in Qt OS plugin. Bummer if so.

**Comment by Hipshot-zz** [comment]
Yea, toolbar has that blue indicator, so that works well. It's just the menu and the menu when it's disconnected from the menu that doesn't have that.

**Comment by Garux** [comment]
https://bugreports.qt.io/browse/QTBUG-98354
damnit, they are not going to fix this in Qt5

**Reply by ensiform** [reply]
Is it fixed in qt6?

**Reply by Garux** [reply]
Says yes
![image](https://github.com/Garux/netradiant-custom/assets/1734117/fb5875d7-d8cd-43e9-ad0b-5fe018fbbb19)

**Comment by Hipshot-zz** [comment]
> https://bugreports.qt.io/browse/QTBUG-98354 damnit, they are not going to fix this in Qt5

I'm not sure what this means, NRC is using QT5 I suppose? Is there a reason to ever change to QT6 (except for this bug then)? QT is just like, the gui, or shell? Like, uhm, BFGX kinda?

**Reply by Garux** [reply]
using QT5, 6 is not wanted yet, as it's win10+
Qt is framework, NRC mainly uses it for GUI and OpenGl

This issue is sad, but not critical, as there is toolbar, which works, and majority prefers dark themes nowadays.
Some idea wrt bright theme is to use qfusion style plugin with default colors instead of vista one. Do buttons have state indication for you with dark theme? (this is qfusion plugin)

**Comment by Hipshot-zz** [comment]
Yes button in the menu have that in dark, in the toolbar both light and dark theme.

But, the entire thing feels a bit unconsistent, in any theme color, because some things have icons and others have checkboxes. Or is it that things that needs to show up in the toolbar needs to have an icon (ofc, what else can be shown so), that's why we have icons with just letters, like W (world), S (structural) and D (detail).

To be honest, it's just a little annoying, it really dosn't effect me that much, I can live wiht this. The white light color is more annoying =)
